CharlieBackend.Business/Services/HomeworkStudentService.cs

```diff
@@ -0,0 +1,215 @@
+﻿using AutoMapper;
+using CharlieBackend.Business.Services.Interfaces;
+using CharlieBackend.Core.DTO.HomeworkStudent;
+using CharlieBackend.Core.Entities;
+using CharlieBackend.Core.Models.ResultModel;
+using CharlieBackend.Data.Repositories.Impl.Interfaces;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CharlieBackend.Business.Services
+{
+    public class HomeworkStudentService : IHomeworkStudentService
```

Comment:
I suggest adding summaries to `HomeworkStudentService` and `HomeworkService` classes in order to avoid confusion in the future. By reading summary we should be able to tell the difference between the services and understand their purpose.

Perhaps something like this
```suggestion
    /// <summary>
    /// Service that provides methods for student's actions with Homework
    /// </summary>
    public class HomeworkStudentService : IHomeworkStudentService
```

And then summary for `HomeworkService` could look something like this:
```cs
    /// <summary>
    /// Service that provides methods for administrative and mentor's work 
    /// with Homeworks
    /// </summary>
    public class HomeworkService : IHomeworkService
```